### ISSUE TITLE
Flower/insects will stop bot

### ISSUE THREAD

**Issue by Ronso20** [fishyboteso/fishyboteso#84]
Flower or insect get's in the way, bot doesn't know how to respond. Is it possible to learn it so it will just skip it and continue in path ?

**Comment by adsau59** [comment]
can you provide a video of when this issue occurs?

**Comment by Ronso20** [comment]
I tried to make a video where this happens. It seems, it gets stuck only if something spawn in front of you while bot is fishing.

In that simulation a made record where bot goes to the fishing hole and start fishing. Then i move camera little bit down to simulate spawned harvestable object. Bot will be staying here until someone harvest it or you move camera manually.

https://youtu.be/nuhOvbuuAfE

**Comment by wabkia** [comment]
Yes, this same thing happens with my defense combat routine. After killing something if my couple of key presses don’t get the loot then the bot literally “freezes”. I think it is waiting for a state change but unless there is combat the state will never change. I wanted to see what would happen and it simply did nothing until i hit the loot key, where it started to fish again. It’s 100% reproducible. I’m not sure if it’s to do with the state getting stuck or what cause it seems like the issue is the bot knows something is there to be ‘interacted’ with but since it’s not a fishing hole it just sits there waiting to hit the interact key. I’m very open to ideas. I think a timeout timer in the right place will solve this temporarily but I’m not sure how to implement it.

**Comment by adsau59** [comment]
> 
> [youtu.be/nuhOvbuuAfE](https://youtu.be/nuhOvbuuAfE)
> 
@Ronso20 im not sure what u are showing in the video

**Comment by adsau59** [comment]
@wabkia can u provide a video reproducing the issue?
